gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -297,7 +297,13 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                 classOutputDir,
                 resourceOutputDir
             )
-            kotlinCompilation.allKotlinSourceSets.forEach { sourceSet -> kspTask.source(sourceSet.kotlin) }
+            kspTask.source(kotlinCompileTask.source)
```

Comment:
This may be tricky as kotlin compile task will have java/kotlin sources that are generated by ksp processors. In `kaptGenerateStubsTask` this is avoided by filtering out sources that are in the task output directories.

What is the motivation for this change, typically sourceSets should contain all information, and plugins should avoid adding sources directly to the task?

Reply:
I was thinking about minimizing surprises for users moving from kapt. This particular case / bug report is about sources added via `BaseVariant.registerJavaGeneratingTask` by other plugins.

Having said that, this behavior isn't quite right. We should probably not picking up generated sources at all.

Reply:
With the toggle, this LGTM. We can also follow up with users that do enable it, to understand their use-cases a bit better.